Doc/library/functools.rst

```diff
@@ -57,6 +57,10 @@ The :mod:`functools` module defines the following functions:
    another thread makes an additional call before the initial call has been
    completed and cached.
 
+   Call-once behavior is not guaranteed: no locks are held during the function
+   call, so the same key may be computed multiple times concurrently.
+
```

Comment:
That is pretty good.  Suggest a minor edit to:

> Call-once behavior is not guaranteed because locks are not held during the function call. Potentially another call with the same arguments could occur before the first call has returned.